packages/babel-plugin-proposal-private-property-in-object/src/index.ts

```diff
@@ -6,9 +6,16 @@ import {
   injectInitialization as injectConstructorInit,
 } from "@babel/helper-create-class-features-plugin";
 import annotateAsPure from "@babel/helper-annotate-as-pure";
+import type * as t from "@babel/types";
+import type { NodePath } from "@babel/traverse";
 
-export default declare(({ assertVersion, types: t, template }, { loose }) => {
-  assertVersion(7);
+export interface Options {
+  loose?: boolean;
+}
+export default declare((api, opt: Options) => {
+  api.assertVersion(7);
+  const { types: t, template } = api;
```

Comment:
Q: Does TS complain about destructuring?

Reply:
Nope. TS correctly picks up `typeof import ("@babel/types")` for `t`.

Reply:
I didn't select enough lines when commenting! The original code was export default declare(({ assertVersion, types: t, template }`, and I wondered if the destructuring directly in the param made it complain.

Reply:
> destructuring directly in the param made it complain

Nope. I moved destructuring in param to body in order to avoid a wall of whitespace changes. In `packages/babel-plugin-proposal-class-static-block/src/index.ts` destructuring in param works as expected.